pkg/v1/google/auth.go

```diff
@@ -115,6 +127,11 @@ func (gs gcloudSource) Token() (*oauth2.Token, error) {
 	// Don't attempt to interpret stderr, just pass it through.
 	cmd.Stderr = os.Stderr
 
+	if _, err := exec.LookPath("gcloud"); err != nil {
+		// gcloud is not available, fall back to anonymous
+		return nil, fallthroughErr{msg: "`gcloud` binary not found"}
```

Comment:
What about just returning `authn.Anonymous, nil` here instead of a new error? Then I think you wouldn't need any other code changes... WDYT?

Reply:
@jonjohnsonjr That was my first take, and the simplest :sweat_smile: But this means using `google.Keychain` alone will fail silently if `gcloud` is missing — I wasn't sure if that was what we want :angel: 

I can definitely update the PR to `return authn.Anonymous, nil` instead :wink:

Reply:
I'd expect users of google.Keychain to have things set up appropriately. We don't have a great story for signaling non-fatal warnings (#390) unfortunately :/

I think the current behavior of returning Anonymous if the keychain failed to find a match (even due to error) is the best we can do for now. I wouldn't be opposed to adding something like:
```go
// TODO(#390): Use better logger.
log.Println("gcloud binary not found")
```

I don't think many people use `google.Keychain`, and those that do probably wouldn't mind that warning message for now 😄

Reply:
alright, let's go for that then :wink: